internal/repositories/vehicle/vehicles_test.go

```diff
@@ -979,3 +997,247 @@ func (o *AccessibleVehiclesRepoTestSuite) Test_GetAccessibleVehicles_Pagination_
+				Owner: wallet1,
+			},
+			results: []*gmodel.VehicleEdge{
+				{Node: vehicle1AsAPI},
+				{Node: vehicle2AsAPI},
+			},
+		},
+		{
+			name: "Filter by Privileged",
+			filter: &gmodel.VehiclesFilter{
+				Privileged: wallet1,
+			},
+			results: []*gmodel.VehicleEdge{
+				{Node: vehicle1AsAPI},
+				{Node: vehicle2AsAPI},
+				{Node: vehicle3AsAPI},
+			},
+		},
+		{
+			name: "Filter by Make",
+			filter: &gmodel.VehiclesFilter{
+				Make: toyota.Ptr(),
+			},
+			results: []*gmodel.VehicleEdge{
+				{Node: vehicle1AsAPI},
+				{Node: vehicle3AsAPI},
+			},
+		},
+		{
+			name: "Filter by Model",
+			filter: &gmodel.VehiclesFilter{
+				Model: camry.Ptr(),
+			},
+			results: []*gmodel.VehicleEdge{
+				{Node: vehicle1AsAPI},
+			},
+		},
+		{
+			name: "Filter by Year",
+			filter: &gmodel.VehiclesFilter{
+				Year: year2022.Ptr(),
+			},
+			results: []*gmodel.VehicleEdge{
+				{Node: vehicle2AsAPI},
+				{Node: vehicle3AsAPI},
+			},
+		},
+
+		{
+			name: "Filter by Owner and Privileged same address",
+			filter: &gmodel.VehiclesFilter{
+				Privileged: wallet1,
+				Owner:      wallet1,
+			},
+			results: []*gmodel.VehicleEdge{
+				{Node: vehicle1AsAPI},
+				{Node: vehicle2AsAPI},
+			},
+		},
+		{
+			name: "Filter by Owner and Privileged different addresses",
+			filter: &gmodel.VehiclesFilter{
+				Privileged: wallet1,
+				Owner:      wallet2,
+			},
+			results: []*gmodel.VehicleEdge{
+				{Node: vehicle3AsAPI},
+			},
+		},
+		{
+			name: "Filter by Privileged and Make",
+			filter: &gmodel.VehiclesFilter{
+				Privileged: wallet1,
+				Make:       toyota.Ptr(),
+			},
+			results: []*gmodel.VehicleEdge{
+				{Node: vehicle1AsAPI},
+				{Node: vehicle3AsAPI},
+			},
+		},
+		{
+			name: "Filter by Privileged and Model",
+			filter: &gmodel.VehiclesFilter{
+				Privileged: wallet1,
+				Model:      camry.Ptr(),
+			},
+			results: []*gmodel.VehicleEdge{
+				{Node: vehicle1AsAPI},
+			},
+		},
+		{
+			name: "Filter by Privileged and Year",
+			filter: &gmodel.VehiclesFilter{
+				Privileged: wallet1,
+				Year:       year2022.Ptr(),
+			},
+			results: []*gmodel.VehicleEdge{
+				{Node: vehicle2AsAPI},
+				{Node: vehicle3AsAPI},
+			},
+		},
+		{
+			name: "Filter No Results",
+			filter: &gmodel.VehiclesFilter{
+				Owner:      wallet1,
+				Privileged: wallet2,
+			},
+			results: []*gmodel.VehicleEdge{},
+		},
+	}
+
+	for i := range tests {
+		tt := tests[i]
+		o.Run(tt.name, func() {
+			res, err := o.repo.GetVehicles(o.ctx, &first, nil, nil, nil, tt.filter)
+			o.Require().NoError(err)
+			o.Require().NotNil(res)
+			requireEqualVehicles(o.T(), tt.results, res.Edges)
+			o.Require().Equalf(len(tt.results), res.TotalCount, "Test %s: expected total count to be %d, got %d", tt.name, len(tt.results), res.TotalCount)
+		})
+	}
+}
+
+// requireEqualVehicles is a helper function to compare two slices of VehicleEdges
+func requireEqualVehicles(t *testing.T, expected, actual []*gmodel.VehicleEdge) {
```

Comment:
I'm not sure it makes sense in this case (are the cursors going to mess it up?), but note that there is 

https://pkg.go.dev/github.com/stretchr/testify/assert#ElementsMatch

Reply:
There 2 reasons for this functions were.
1. The curosors mess this up.
2. ElementsMatch doesn't work as desired because it does not derefrence the element, thus it is looking for identical adresses.

Reply:
I can never predict when these libraries are going to follow pointers.